### DISCUSSION TITLE
r.learn.ml2 - train-test-split and access to all scikit-learn parameters?

### DISCUSSION THREAD

**Discussion by CarlosGrohmann** [OSGeo/grass#1859] [Q&A] [answered]
I started to test `r.learn.ml2`. It worked fine on my first attempt. 

But I'm wondering how to do a train-test split and if there is a way to define values for all parameters that can be used in scikit-learn (for example, `RandomForestClassifier` has a "criterion" parameter).

thanks

**Comment by stevenpawley** [comment, marked as answer]
Hi Carlos,

Right now, the r.learn.ml2 provides only the most commonly used hyper parameters for each model type - simplified for GUI users. If you are wanting to perform a more detailed data science workflow, I would suggest that you import the modules directly within the python script. There is some information https://stevenpawley.github.io/r.learn.ml2/python_quickstart.html relating to this.

However, I'm not doing much work on r.learn.ml2 at the moment. You may find my python package https://pyspatialml.readthedocs.io/en/latest/ more useful in this respect, which has seen a reasonable uptake in terms of applying ML models to spatial data, in a similar style to 'raster' in R.

Steve

**Reply by CarlosGrohmann** [reply]
Thanks for the quick reply Steven 

If I'm interested in the classification of satellite images, would you say that a good approach would be to use GRASS for the pre-processing (conversion to reflectance, atmospheric correction, clipping), export as TIFs, run ML with pyspatialml (then re-importing into GRASS)? 

Carlos

**Reply by stevenpawley** [reply]
Hmm, I think that depends on your workflow. If you have large volumes of data already stored in GRASS, then r.learn.ml2 is the way to go. Also, if your imagery is not fully aligned and you just want to 'drop in' to use existing datasets for analysis, GRASS obviously has lots of advantages compared to the typical R/Python approach where you create a fully cleaned/aligned and self-contained project.

**Reply by CarlosGrohmann** [reply]
Thanks!
